src/stats.c

```diff
@@ -995,16 +1025,17 @@ stats_general_print(void (*write_cb)(void *, const char *), void *cbopaque,
 
 static void
 read_global_mutex_stats(
-    uint64_t results[NUM_GLOBAL_PROF_MUTEXES][NUM_MUTEX_PROF_COUNTERS]) {
+    uint64_t results[num_global_prof_mutexes][num_mutex_prof_counters]) {
 	char cmd[128];
 
-	unsigned i, j;
-	for (i = 0; i < NUM_GLOBAL_PROF_MUTEXES; i++) {
-		for (j = 0; j < NUM_MUTEX_PROF_COUNTERS; j++) {
-			gen_mutex_ctl_str(cmd, "mutexes", global_mutex_names[i],
-			    mutex_counter_names[j]);
-			CTL_GET(cmd, &results[i][j], uint64_t);
-		}
+	global_prof_mutex_ind_t i;
+	for (i = 0; i < num_global_prof_mutexes; i++) {
+#define OP(c, t)							\
+		gen_mutex_ctl_str(cmd, 128, "mutexes",			\
```

Comment:
This is brittle because a bare magic number has to be kept consistent in multiple places.  I don't feel strongly enough about the ```128``` scattered around to force the issue, but you might consider adding a ```#define``` for these to make it easier to find the related magic numbers during future refactoring.